addons/mrp/tests/test_order.py

```diff
@@ -894,3 +894,109 @@ def test_product_produce_uom(self):
+            'category_id': categ_test.id,
+            'uom_type': 'reference',
+            'rounding': 0.001
+        })
+
+        uom_ml = self.env['uom.uom'].create({
+            'name': 'Test ml',
+            'category_id': categ_test.id,
+            'uom_type': 'smaller',
+            'rounding': 0.01,
+            'factor_inv': 0.001,
+        })
+
+        # create a product component and the final product using the component
+        product_comp = self.env['product.product'].create({
+            'name': 'Product Component',
+            'type': 'product',
+            'tracking': 'lot',
+            'categ_id': self.env.ref('product.product_category_all').id,
+            'uom_id': uom_L.id,
+            'uom_po_id': uom_L.id,
+        })
+
+        product_final = self.env['product.product'].create({
+            'name': 'Product Final',
+            'type': 'product',
+            'tracking': 'lot',
+            'categ_id': self.env.ref('product.product_category_all').id,
+            'uom_id': uom_L.id,
+            'uom_po_id': uom_L.id,
+        })
+
+        # the products are tracked by lot, so we go through _generate_consumed_move_line
+        lot_final = self.env['stock.production.lot'].create({
+            'name': 'Lot Final',
+            'product_id': product_final.id,
+        })
+
+        lot_comp = self.env['stock.production.lot'].create({
+            'name': 'Lot Component',
+            'product_id': product_comp.id,
+        })
+
+        # update the quantity on hand for Component, in a lot
+        self.stock_location = self.env.ref('stock.stock_location_stock')
+        self.env['stock.quant']._update_available_quantity(product_comp, self.stock_location, 1, lot_id=lot_comp)
+
+        # create a BOM for Final, using Component
+        test_bom = self.env['mrp.bom'].create({
+            'product_id': product_final.id,
+            'product_tmpl_id': product_final.product_tmpl_id.id,
+            'product_uom_id': uom_L.id,
+            'product_qty': 1.0,
+            'type': 'normal',
+            'bom_line_ids': [(0, 0, {
+                'product_id': product_comp.id,
+                'product_qty': 375.00,
+                'product_uom_id': uom_ml.id
+            })],
+        })
+
+        # create a MO for this BOM
+        mo_product_final = self.env['mrp.production'].create({
+            'product_id': product_final.id,
+            'product_qty': 0.5,
+            'product_uom_id': uom_L.id,
+            'bom_id': test_bom.id,
+        })
+        mo_product_final.action_assign()
+        self.assertEqual(mo_product_final.availability, 'assigned')
+
+        # produce
+        context = {"active_ids": [mo_product_final.id], "active_id": mo_product_final.id}
+        product_produce = self.env['mrp.product.produce'].with_context(context).create({
+            'product_qty': 0.5,
+            'lot_id': lot_final.id,
+            })
+        product_produce._onchange_product_qty()
```

Comment:
@amoyaux updated this test, does the rest look ok to you?

Reply:
You should use Form obejct. instead of calling onchange https://github.com/odoo/odoo/blob/12.0/odoo/tests/common.py#L1075